### ISSUE TITLE
[NONROOT] Install loads indefinitely

### ISSUE THREAD

**Issue by RudiAusBuddeln** [YTVanced/VancedManager#324]
**Phone Specifications:**
- Brand: Samsung Tablet SM-T530
- Operating System: Android (Linux 3.4.0)
- Android Version: 5.0.2
- Vanced Manager Version: 2.1.0

**Please describe the problem you are having in as much detail as possible:**

First, updating Youtube Vanced through the manager didn't work; instead it loaded indefinitely. So I uninstalled Youtube Vanced and then attempted to install it freshly through the manager, but the same problem occurred: after downloading the files, the popup "Youtube Vanced – Please wait" just stays there with no progress. Furthermore the cancel button ceases to work, moreso the entire manager app becomes frozen and must be closed externally.
I'd gladly send a log, but I didn't find an option to view them in the app.

**Steps to reproduce:**

On my device, 

**Further details:**

This is not a new notion with an update, but the manager app never actually worked. 

It also looks similar to #235 however my device is not rooted.

I also cannot install Youtube Vanced directly from an APK, since it always fails with no error description.

**Comment by X1nto** [comment]
A logcat would be helpful in this situation

**Comment by RudiAusBuddeln** [comment]
Gladly, but how do I provide one? The app didn't give an option to open the logs folder or similar.

**Comment by mydongistiny** [comment]
> Gladly, but how do I provide one? The app didn't give an option to open the logs folder or similar.

Either install adb for your operating system and run
`adb logcat`
or install a logcat app like Matlog from the Play Store.

**Comment by RudiAusBuddeln** [comment]
My device isn't rooted, so I used `adb`. I have never used it though, so I found the following command to get the PID of a single app, after I forced the manager to shutdown and opened it again:

```
pid=$(adb shell ps | grep "com.vanced.manager" | cut -c11-15)
adb logcat | grep $pid
```

Then I simply clicked on update and install since the files are already downloaded. I stayed at line 105 for a couple of minutes, then the tablet went into sleep mode. Then it stayed at the end of the log until I forced the app to shut down.

```
E/Zygote  (15417): MountEmulatedStorage()
E/Zygote  (15417): v2
I/libpersona(15417): KNOX_SDCARD checking this for 10098
I/libpersona(15417): KNOX_SDCARD not a persona
I/SELinux (15417): Function: selinux_compare_spd_ram, SPD-policy is existed. and_ver=SEPF_SM-T530_5.0.2 ver=38
I/SELinux (15417): Function: selinux_compare_spd_ram , priority [2] , priority version is VE=SEPF_SM-T530_5.0.2-1_0072
E/SELinux (15417): [DEBUG] get_category: variable seinfo: default sensitivity: NULL, cateogry: NULL
I/ActivityManager(  918): Start proc com.vanced.manager for activity com.vanced.manager/.ui.core.SplashScreenActivity: pid=15417 uid=10098 gids={50098, 9997, 3003, 1028, 1015} abi=armeabi-v7a
D/ActivityThread(15417): Added TimaKeyStore provider
W/ComponentDiscovery(15417): Class com.google.firebase.dynamicloading.DynamicLoadingRegistrar is not an found.
I/FirebaseApp(15417): Device unlocked: initializing all Firebase APIs for app [DEFAULT]
I/DynamiteModule(15417): Considering local module com.google.android.gms.measurement.dynamite:20 and remote module com.google.android.gms.measurement.dynamite:0
I/DynamiteModule(15417): Selected local version of com.google.android.gms.measurement.dynamite
W/FA      (15417): Failed to register event listener on calling thread. Trying again on the dynamite thread.
I/FirebaseCrashlytics(15417): Initializing Crashlytics 17.3.0
I/FirebaseInitProvider(15417): FirebaseApp initialization successful
E/SQLiteLog(15417): (283) recovered 27 frames from WAL file /data/data/com.vanced.manager/databases/androidx.work.workdb-wal
W/ActivityManager(  918): Permission Denial: getCurrentUser() from pid=15417, uid=10098 requires android.permission.INTERACT_ACROSS_USERS
I/art     (15417): Rejecting re-init on previously-failed class java.lang.Class<r.e.a.e.e>
I/art     (15417): Rejecting re-init on previously-failed class java.lang.Class<r.e.a.e.e>
E/File    (15417): fail readDirectory() errno=2
E/File    (15417): fail readDirectory() errno=2
I/System.out(15417): (HTTPLog)-Static: SBServiceAPI: getService class android.os.ServiceManager
I/System.out(15417): (HTTPLog)-Static: isSBSettingEnabled false
I/System.out(15417): (HTTPLog)-Static: isShipBuild true
I/System.out(15417): (HTTPLog)-Thread-2133-348439025: SmartBonding Enabling is false, SHIP_BUILD is true, log to file is false, DBG is false
I/System.out(15417): (HTTPLog)-Static: isSBSettingEnabled false
I/System.out(15417): KnoxVpnUidStorageknoxVpnSupported API value returned is false
I/FA      (15417): App measurement disabled by setAnalyticsCollectionEnabled(false)
I/FA      (15417): App measurement initialized, version: 33025
I/FA      (15417): To enable debug logging run: adb shell setprop log.tag.FA VERBOSE
I/FA      (15417): To enable faster debug mode event logging run:
I/FA      (15417):   adb shell setprop debug.firebase.analytics.app com.vanced.manager
W/ResourceType(15417): Failure getting entry for 0x01080ad0 (t=7 e=2768) (error -75)
I/System.out(15417): (HTTPLog)-Static: isSBSettingEnabled false
I/art     (15417): Background partial concurrent mark sweep GC freed 26803(1553KB) AllocSpace objects, 5(174KB) LOS objects, 74% free, 5MB/21MB, paused 4.476ms total 146.194ms
W/ActivityManager(  918): Permission Denial: getCurrentUser() from pid=15417, uid=10098 requires android.permission.INTERACT_ACROSS_USERS
I/Timeline(15417): Timeline: Activity_launch_request id:com.vanced.manager time:23601394
W/ResourceType(15417): Failure getting entry for 0x01080ad0 (t=7 e=2768) (error -75)
W/ResourceType(15417): Failure getting entry for 0x01080ad1 (t=7 e=2769) (error -75)
W/ActivityManager(  918): Permission Denial: getCurrentUser() from pid=15417, uid=10098 requires android.permission.INTERACT_ACROSS_USERS
D/PhoneWindow(15417): *FMB* installDecor mIsFloating : false
D/PhoneWindow(15417): *FMB* installDecor flags : -2139029248
W/ResourceType(15417): Failure getting entry for 0x01080ad1 (t=7 e=2769) (error -75)
D/OpenGLRenderer(15417): Render dirty regions requested: true
D/PhoneWindow(15417): *FMB* isFloatingMenuEnabled mFloatingMenuBtn : null
D/PhoneWindow(15417): *FMB* isFloatingMenuEnabled return false
D/InputDispatcher(  918): Focus entered window: 15417
I/Adreno-EGL(15417): <qeglDrvAPI_eglInitialize:410>: EGL 1.4 QUALCOMM build:  ()
I/Adreno-EGL(15417): OpenGL ES Shader Compiler Version: E031.25.01.03
I/Adreno-EGL(15417): Build Date: 03/13/15 Fri
I/Adreno-EGL(15417): Local Branch: LA.BF.1.1.2_RB1-AU004-20150313-1115765-1110870-1156580
I/Adreno-EGL(15417): Remote Branch: 
I/Adreno-EGL(15417): Local Patches: 
I/Adreno-EGL(15417): Reconstruct Branch: 
I/OpenGLRenderer(15417): Initialized EGL, version 1.4
D/OpenGLRenderer(15417): Get maximum texture size. GL_MAX_TEXTURE_SIZE is 4096
D/OpenGLRenderer(15417): Enabling debug mode 0
W/PathParser(15417): Points are too far apart 4.000001856601723
W/PathParser(15417): Points are too far apart 4.000001856601723
D/VMLocalisation(15417): Failed to load data: java.lang.Throwable: Network operation failed 403
I/Timeline(15417): Timeline: Activity_idle id: android.os.BinderProxy@14bee33 time:23602503
W/ConnectionTracker(15417): Exception thrown while unbinding
W/ConnectionTracker(15417): java.lang.IllegalArgumentException: Service not registered: r.f.a.c.i.b.m8@32191608
W/ConnectionTracker(15417):     at android.app.LoadedApk.forgetServiceDispatcher(LoadedApk.java:1096)
W/ConnectionTracker(15417):     at android.app.ContextImpl.unbindService(ContextImpl.java:2188)
W/ConnectionTracker(15417):     at android.content.ContextWrapper.unbindService(ContextWrapper.java:571)
W/ConnectionTracker(15417):     at r.f.a.c.e.p.a.c(:3)
W/ConnectionTracker(15417):     at r.f.a.c.i.b.s7.D(:5)
W/ConnectionTracker(15417):     at r.f.a.c.i.b.r7.a(:3)
W/ConnectionTracker(15417):     at r.f.a.c.i.b.k.run(:4)
W/ConnectionTracker(15417):     at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:422)
W/ConnectionTracker(15417):     at java.util.concurrent.FutureTask.run(FutureTask.java:237)
W/ConnectionTracker(15417):     at r.f.a.c.i.b.x4.run(:2)
I/InputDispatcher(  918): Delivering touch to (15417): action: 0x0, toolType: 1
D/ViewRootImpl(15417): ViewPostImeInputStage ACTION_DOWN
I/InputDispatcher(  918): Delivering touch to (15417): action: 0x1, toolType: 1
D/PhoneWindow(15417): *FMB* installDecor mIsFloating : true
D/PhoneWindow(15417): *FMB* installDecor flags : 8388610
I/AppCompatDelegate(15417): The Activity's LayoutInflater already has a Factory installed so we can not install AppCompat's
D/InputDispatcher(  918): Focus left window: 15417
D/InputDispatcher(  918): Focus entered window: 15417
D/PhoneWindow(15417): *FMB* isFloatingMenuEnabled mFloatingMenuBtn : null
D/PhoneWindow(15417): *FMB* isFloatingMenuEnabled return false
I/InputDispatcher(  918): Delivering touch to (15417): action: 0x0, toolType: 1
D/ViewRootImpl(15417): ViewPostImeInputStage ACTION_DOWN
I/InputDispatcher(  918): Delivering touch to (15417): action: 0x1, toolType: 1
D/InputDispatcher(  918): Focus left window: 15417
D/InputDispatcher(  918): Focus entered window: 15417
E/ViewRootImpl(15417): sendUserActionEvent() mView == null
I/InputDispatcher(  918): Delivering touch to (15417): action: 0x0, toolType: 1
D/ViewRootImpl(15417): ViewPostImeInputStage ACTION_DOWN
I/InputDispatcher(  918): Delivering touch to (15417): action: 0x1, toolType: 1
D/PhoneWindow(15417): *FMB* installDecor mIsFloating : true
D/PhoneWindow(15417): *FMB* installDecor flags : 8388610
I/AppCompatDelegate(15417): The Activity's LayoutInflater already has a Factory installed so we can not install AppCompat's
D/InputDispatcher(  918): Focus left window: 15417
D/InputDispatcher(  918): Focus entered window: 15417
D/PhoneWindow(15417): *FMB* isFloatingMenuEnabled mFloatingMenuBtn : null
D/PhoneWindow(15417): *FMB* isFloatingMenuEnabled return false
I/InputDispatcher(  918): Delivering touch to (15417): action: 0x0, toolType: 1
D/ViewRootImpl(15417): ViewPostImeInputStage ACTION_DOWN
I/InputDispatcher(  918): Delivering touch to (15417): action: 0x1, toolType: 1
D/InputDispatcher(  918): Focus left window: 15417
D/InputDispatcher(  918): Focus entered window: 15417
D/AppLog  (15417): installApk: dark.apk
D/AppLog  (15417): installApk: split_config.armeabi_v7a.apk
D/AppLog  (15417): installApk: split_config.en.apk
D/AppLog  (15417): installApk: split_config.es.apk
D/AppLog  (15417): Success: created install session [662852287]
D/AppLog  (15417): Success: streamed 13281217 bytes
D/AppLog  (15417): Success: streamed 205178 bytes
D/AppLog  (15417): Success: streamed 237946 bytes
D/AppLog  (15417): Success: streamed 21717154 bytes
D/AppLog  (15417): install request sent
D/AppLog  (15417): doCommitSession: [android.content.pm.PackageInstaller$SessionInfo@37815b35, android.content.pm.PackageInstaller$SessionInfo@bdc2bca]
D/AppLog  (15417): doCommitSession: after session commit 
D/AppLog  (15417): Success
I/Choreographer(15417): Skipped 430 frames!  The application may be doing too much work on its main thread.
E/ViewRootImpl(15417): sendUserActionEvent() mView == null
W/art     (15417): Before Android 4.1, method android.graphics.drawable.Drawable r.f.a.d.u.b.getCurrentDrawable() would have incorrectly overridden the package-private method in android.widget.ProgressBar
D/PhoneWindow(15417): *FMB* installDecor mIsFloating : true
D/PhoneWindow(15417): *FMB* installDecor flags : 8388610
I/AppCompatDelegate(15417): The Activity's LayoutInflater already has a Factory installed so we can not install AppCompat's
D/InputDispatcher(  918): Focus left window: 15417
D/InputDispatcher(  918): Focus entered window: 15417
D/VMInstall(15417): Requesting user confirmation for installation
D/Instrumentation(15417): checkStartActivityResult  :Intent { act=android.content.pm.action.CONFIRM_PERMISSIONS flg=0x10000000 pkg=com.android.packageinstaller (has extras) }
D/Instrumentation(15417): checkStartActivityResult  inent is instance of inent:
D/VMInstall(15417): Unable to start installation
D/PhoneWindow(15417): *FMB* isFloatingMenuEnabled mFloatingMenuBtn : null
D/PhoneWindow(15417): *FMB* isFloatingMenuEnabled return false
I/InputDispatcher(  918): Delivering touch to (15417): action: 0x0, toolType: 1
D/ViewRootImpl(15417): ViewPostImeInputStage ACTION_DOWN
I/InputDispatcher(  918): Delivering touch to (15417): action: 0x1, toolType: 1
I/InputDispatcher(  918): Delivering touch to (15417): action: 0x0, toolType: 1
D/ViewRootImpl(15417): ViewPostImeInputStage ACTION_DOWN
I/InputDispatcher(  918): Delivering touch to (15417): action: 0x1, toolType: 1
V/ActivityThread(15417): updateVisibility : ActivityRecord{ddc499f token=android.os.BinderProxy@14bee33 {com.vanced.manager/com.vanced.manager.ui.MainActivity}} show : true
D/InputDispatcher(  918): Focus left window: 15417
D/InputDispatcher(  918): Focus entered window: 15417
W/ConnectionTracker(15417): Exception thrown while unbinding
W/ConnectionTracker(15417): java.lang.IllegalArgumentException: Service not registered: r.f.a.c.i.b.m8@32191608
W/ConnectionTracker(15417):     at android.app.LoadedApk.forgetServiceDispatcher(LoadedApk.java:1096)
W/ConnectionTracker(15417):     at android.app.ContextImpl.unbindService(ContextImpl.java:2188)
W/ConnectionTracker(15417):     at android.content.ContextWrapper.unbindService(ContextWrapper.java:571)
W/ConnectionTracker(15417):     at r.f.a.c.e.p.a.c(:3)
W/ConnectionTracker(15417):     at r.f.a.c.i.b.s7.D(:5)
W/ConnectionTracker(15417):     at r.f.a.c.i.b.r7.a(:3)
W/ConnectionTracker(15417):     at r.f.a.c.i.b.k.run(:4)
W/ConnectionTracker(15417):     at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:422)
W/ConnectionTracker(15417):     at java.util.concurrent.FutureTask.run(FutureTask.java:237)
W/ConnectionTracker(15417):     at r.f.a.c.i.b.x4.run(:2)
D/InputDispatcher(  918): Focus left window: 15417
D/InputDispatcher(  918): Focus entered window: 15417
W/ResourceType(15417): Failure getting entry for 0x01080ad1 (t=7 e=2769) (error -75)
I/Timeline(15417): Timeline: Activity_idle id: android.os.BinderProxy@14bee33 time:23802991
I/InputDispatcher(  918): Delivering touch to (15417): action: 0x0, toolType: 1
D/ViewRootImpl(15417): ViewPostImeInputS
```

**Comment by X1nto** [comment]
@RudiAusBuddeln did you allow installations from unknown sources? also, try granting storage permissions to manager and then install again

**Comment by X1nto** [comment]
Should be fixed in 2.4.0. reopen issue if it's not
